src/Ocelot/DynamicConfigurationProvider/DynamicConfigurationProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Ocelot.Configuration;
+using Ocelot.Configuration.File;
+using Ocelot.Logging;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Ocelot.DynamicConfigurationProvider
+{
+    public abstract class DynamicConfigurationProvider
```

Comment:
Added base class specially for diagnostic logs, because that would always allow to check for latency in case of custom provider implementation. But, yes, even i prefer interfaces over classes.